designs/2019-config-simplification/README.md

```diff
@@ -0,0 +1,544 @@
+
+Although `.eslintrc` has proven to be a very robust format, it is also limited and unable to easily accommodate feature requests that the community favors, such as:
+
+1. [bundling plugins with configs](https://github.com/eslint/eslint/issues/3458)
+1. [specifying ignore patterns in the config](https://github.com/eslint/eslint/issues/10891)
+1. [specifying `--ext` information in the config](https://github.com/eslint/eslint/issues/11223)
+1. [using `extends` in `overrides`](https://github.com/eslint/eslint/issues/8813) 
+
+The only reason that these requests are difficult to implement in ESLint is because of how complex the `.eslintrc` configuration format is. Any changes made to any part of `.eslintrc` processing end up affecting millions of ESLint installations, so we have ended up stuck. The complicated parts of `.eslintrc` include:
+
+1. Resolution behavior of modules (`plugins`, `parser`, `extends`)
+1. Merging of cascading config files in a directory structure
+1. Merging of config files via `extends`
+1. Overriding of configuration via `overrides`
+
+All of these complexities arise from `.eslintrc` format of describing what should happen rather than how it should happen. We keep trying to anticipate how users want things to happen and guessing wrong has led to increased complexity.
+
+This RFC proposes a fresh start for configuration in ESLint that takes into account all of the requests we've received over the years. By simplifying and stripping configuration we can actually make a configuration system that is flexible enough to accomodate changes in the future and easy enough to implement that we won't be afraid to do so.
+
+## Detailed Design
+
+Design Summary:
+
+1. Introduce a new `.eslint.js` configuration file format
+1. Searching for `.eslint.js` starts from the current working directory and continues up
+1. All `.eslint.js` files are treated as if they have `root: true`
+1. There is no automatic merging of config files
+1. The `.eslint.js` configuration is not serializable and objects such as functions and plugins may be added directly into configuration
+1. Remove the concept of environments (`env`)
+1. Remove `.eslintignore` and `--ignore-file` (no longer necessary)
+1. Remove `--rulesdir`
+1. Create a `@eslint/config` package to help users merge configs
+
+### The `.eslint.js` File
+
+The `.eslint.js` file is a JavaScript file (there is no JSON or YAML equivalent) that exports a `config` object: 
+
+```js
+exports.config = {
+    files: "*.js",
+    ignore: "*.test.js",
```

Comment:
`files` and `ignore` should be `Array` right? (based on current functionality)

Reply:
I was imagining these being strings that used `;` as a separator. I don't know that having them be arrays gives much of an advantage, but let's see what others think.

Reply:
Letting it be an array means that it can be modified with general JavaScript array methods (like filtering based on environment, concatenation based on multiple child configs, etc.) Also easier to inspect or validate, and will be better formatted by autoformatters like Prettier.

Reply:
These have been changed to arrays per the feedback.